text/0436-gamelift-l2.md

```diff
@@ -0,0 +1,999 @@
+
+There are two options for setting up event notifications. You can set up an SNS topic and have GameLift publish event notifications on placement activity by referencing the topic ID in a game session queue. Alternatively, you can use Amazon CloudWatch Events, which has a suite of tools available for managing events and taking action on them.
+
+
+```ts fixture=with-build
+import * as gamelift from 'aws-cdk-lib/aws-gamelift';
+
+const fleet = new gamelift.Fleet(this, 'Game server fleet', {
+  build: build
+});
+
+const topic = new sns.Topic(this, 'Topic');
+
+const queue = new gamelift.Queue(this, 'Game session queue', {
+  notification: new SnsDestination(topic)
+});
+queue.addDestination(fleet);
+```
+
+##### Managing game servers launch configuration
+
+GameLift uses a fleet's runtime configuration to determine the type and number of processes to run on each instance in the fleet. At a minimum, a runtime configuration contains one server process configuration that represents one game server executable. You can also define additional server process configurations to run other types of processes related to your game. Each server process configuration contains the following information:
+
+* The file name and path of an executable in your game build.
+
+* Optionally Parameters to pass to the process on launch.
+
+* The number of processes to run concurrently.
+
+A GameLift instance is limited to 50 processes running concurrently.
+
+```ts fixture=with-build
+import * as gamelift from '@aws-cdk-lib/aws-gamelift';
+
+const fleet = new gamelift.Fleet(this, 'Game server fleet', {
+  build: build,
+  runtimeConfiguration: {
+    gameSessionActivationTimeoutSeconds: 123,
+    maxConcurrentGameSessionActivations: 123,
+  }
+});
+fleet.runtimeConfiguration.addServerProcess({
+  launchPath: '/local/game/GameLiftExampleServer.x86_64', 
+  parameters: '-logFile /local/game/logs/myserver1935.log -port 1935',
+  concurrentExecutions: 100,
+});
+```
+
+See [Managing how game servers are launched for hosting](https://docs.aws.amazon.com/gamelift/latest/developerguide/fleets-multiprocess.html)
+in the *Amazon GameLift Developer Guide*.
+
+##### Defining an instance type
+
+GameLift uses Amazon Elastic Compute Cloud (Amazon EC2) resources, called instances, to deploy your game servers and host game sessions for your players. When setting up a new fleet, you decide what type of instances your game needs and how to run game server processes on them (using a runtime configuration). All instances in a fleet use the same type of resources and the same runtime configuration. You can edit a fleet's runtime configuration and other fleet properties, but the type of resources cannot be changed.
+
+```ts fixture=with-build
+import * as gamelift from '@aws-cdk-lib/aws-gamelift';
+import * as ec2 from '@aws-cdk-lib/aws-ec2';
+
+new gamelift.Fleet(this, 'Game server fleet', {
+  build: build,
+  instanceType: ec2.InstanceType.of(ec2.InstanceClass.C5, ec2.InstanceSize.LARGE)
```

Comment:
We have a validation on top of that (planned on CDK and also directly on Gamelift API) that's why we want to keep it that way to be compatible more easily with new instances instead of having to release a new version of our CDK Construct.